services/multitenant/tenantstats.go

```diff
@@ -113,95 +116,50 @@ func (multitenantStat *MultitenantStatsT) CalculateSuccessFailureCounts(workspac
-			multitenantStat.routerJobCountMutex.RLock()
+			multitenantStat.routerInputRates[dbPrefix][key] = make(map[string]metric.MovingAverage)
 		}
-		multitenantStat.routerJobCountMutex.RUnlock()
 		for destType := range stats[key] {
-			multitenantStat.routerJobCountMutex.RLock()
-			_, ok := multitenantStat.routerInputRates[tableType][key][destType]
+			_, ok := multitenantStat.routerInputRates[dbPrefix][key][destType]
 			if !ok {
-				multitenantStat.routerJobCountMutex.RUnlock()
-				multitenantStat.routerJobCountMutex.Lock()
-				multitenantStat.routerInputRates[tableType][key][destType] = misc.NewMovingAverage()
-				multitenantStat.routerJobCountMutex.Unlock()
-				multitenantStat.routerJobCountMutex.RLock()
+				multitenantStat.routerInputRates[dbPrefix][key][destType] = metric.NewMovingAverage()
 			}
-			multitenantStat.routerJobCountMutex.RUnlock()
-			multitenantStat.routerInputRates[tableType][key][destType].Add((float64(stats[key][destType]) * float64(time.Second)) / float64(timeTaken))
-			multitenantStat.AddToInMemoryCount(key, destType, stats[key][destType], tableType)
+			multitenantStat.routerInputRates[dbPrefix][key][destType].Add((float64(stats[key][destType]) * float64(time.Second)) / float64(timeTaken))
+			metric.GetPendingEventsMeasurement(dbPrefix, key, destType).Add(float64(stats[key][destType]))
 		}
 	}
-	for workspaceKey := range multitenantStat.routerInputRates[tableType] {
+	for workspaceKey := range multitenantStat.routerInputRates[dbPrefix] {
 		_, ok := stats[workspaceKey]
 		if !ok {
-			for destType := range stats[workspaceKey] {
-				multitenantStat.routerJobCountMutex.Lock()
-				multitenantStat.routerInputRates[tableType][workspaceKey][destType].Add(0)
-				multitenantStat.routerJobCountMutex.Unlock()
+			for destType := range multitenantStat.routerInputRates[workspaceKey] {
+				multitenantStat.routerInputRates[dbPrefix][workspaceKey][destType].Add(0)
 			}
-		}
-
-		for destType := range multitenantStat.routerInputRates[tableType][workspaceKey] {
-			_, ok := stats[workspaceKey][destType]
-			if !ok {
-				multitenantStat.routerJobCountMutex.Lock()
-				multitenantStat.routerInputRates[tableType][workspaceKey][destType].Add(0)
-				multitenantStat.routerJobCountMutex.Unlock()
+		} else {
+			for destType := range multitenantStat.routerInputRates[dbPrefix][workspaceKey] {
+				_, ok := stats[workspaceKey][destType]
+				if !ok {
+					multitenantStat.routerInputRates[dbPrefix][workspaceKey][destType].Add(0)
+				}
 			}
 		}
 	}
 	multitenantStat.processorStageTime = time.Now()
 }
 
-func (multitenantStat *MultitenantStatsT) GetRouterPickupJobs(destType string, noOfWorkers int, routerTimeOut time.Duration, jobQueryBatchSize int, timeGained float64) (map[string]int, map[string]float64) {
+func (multitenantStat *MultitenantStatsT) GetRouterPickupJobs(dbPrefix string, noOfWorkers int, routerTimeOut time.Duration, jobQueryBatchSize int, timeGained float64) (map[string]int, map[string]float64) {
```

Comment:
I think this remains destType
dbPrefix is "rt" for this function anyway - because this is called for router only.
May need to have dbPrefix and destType both if both prefixes call `GetRouterPickupJobs`

Reply:
Correct, I messed this one up. Thanks for spotting it! I will revert and make sure this didn't 
cascade anywhere else.